AmplifyTests/CategoryTests/DataStore/ModelIdentifierTests.swift

```diff
@@ -0,0 +1,126 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import Amplify
+import AmplifyTestCommon
+
+class ModelIdentifierTests: XCTestCase {
+    override func setUp() {
+
+        // default primary keys models
+        ModelRegistry.register(modelType: Post.self)
+
+        // custom primary keys models
+        ModelRegistry.register(modelType: ModelImplicitDefaultPk.self)
+        ModelRegistry.register(modelType: ModelExplicitDefaultPk.self)
+        ModelRegistry.register(modelType: ModelExplicitCustomPk.self)
+
+        ModelRegistry.register(modelType: ModelCompositePk.self)
+
+        // custom pk based on indexes
+        ModelRegistry.register(modelType: CustomerWithMultipleFieldsinPK.self)
+
+        // custom pk based on primaryKey field
+        ModelRegistry.register(modelType: ModelCustomPkDefined.self)
+
+        // legacy pk
+        ModelRegistry.register(modelType: CustomerOrder.self)
+    }
+
+    // MARK: - ModelIdentifier string value
+
+    func testModelIdentifierDefaultId() {
+        let model = Post(title: "title", content: "", createdAt: Temporal.DateTime.now())
+        XCTAssertEqual(model.identifier, model.id)
+    }
+
+    func testModelIdentifierImplicitDefault() {
+        let model = ModelImplicitDefaultPk(name: "name")
+        XCTAssertEqual(model.identifier, model.id)
+    }
+
+    func testModelIdentifierExplicitId() {
+        let model = ModelExplicitDefaultPk(name: "name")
+        XCTAssertEqual(model.identifier, model.id)
+    }
+
+    func testModelIdentifierCustomField() {
+        let model = ModelExplicitCustomPk(userId: "userId")
+        XCTAssertEqual(model.identifier, model.userId)
+    }
+
+    func testModelIdentifierCompositeKey() {
+        let model = ModelCompositePk(id: "id", dob: Temporal.DateTime.now(), name: "name")
+        XCTAssertEqual(model.identifier, "\(model.id)#\(model.dob.iso8601String)")
```

Comment:
Same comment as previous one - shouldn't each field value be in quotes

Reply:
Yes, updated.